index.ts

```diff
@@ -316,68 +362,87 @@ class ServerlessCustomDomain {
     }
 
     /**
-     * Gets domain info as DomainInfo object if domain exists, otherwise returns false
+     * Populates the DomainInfo object on the Domains if custom domain in aws exists
      */
-    public async getDomainInfo(): Promise<DomainInfo> {
-        let domainInfo;
-        try {
-            domainInfo = await this.apigateway.getDomainName({ domainName: this.givenDomainName }).promise();
-            return new DomainInfo(domainInfo);
-        } catch (err) {
-            this.logIfDebug(err);
-            if (err.code === "NotFoundException") {
-                throw new Error(`Error: ${this.givenDomainName} not found.`);
+    public async getDomainInfo(): Promise<void> {
+        await Promise.all(this.domains.map(async (domain) => {
+            try {
+                const domainInfo = await this.apigatewayV2.getDomainName({
```

Comment:
In this case, the v2 API works and I figured it best to use it in as many places as possible in an attempt to standardize the responses and the code that uses these responses. 

The v2 API can read existing Domain and API Mappings without issue. It's only the creation of REST EDGE domains and API Mappings that it can't handle.